src/System.Private.CoreLib/shared/System/Collections/DictionaryEntry.cs

```diff
@@ -12,18 +13,18 @@ namespace System.Collections
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     public struct DictionaryEntry
     {
-        private object _key; // Do not rename (binary serialization)
-        private object _value; // Do not rename (binary serialization)
+        private object? _key; // Do not rename (binary serialization)
+        private object? _value; // Do not rename (binary serialization)
 
         // Constructs a new DictionaryEnumerator by setting the Key
         // and Value fields appropriately.
-        public DictionaryEntry(object key, object value)
+        public DictionaryEntry(object? key, object? value)
```

Comment:
I decided to mark the key as nullable as well since nothing prevents from doing so and anybody can create `DictionaryEntries` as please.